server/bleep/src/webserver/answer.rs

```diff
@@ -596,6 +616,9 @@ struct Agent {
     thread_id: uuid::Uuid,
     query_id: uuid::Uuid,
 
+    query: String,
```

Comment:
Do we actually need to record this at all? We have the `branch` already, can't we just add that in `semantic_query_params`?

Or, if we do want to include the other fields like language and path, can we store those directly like `branch`?

Reply:
The idea would have been to record _all_ filters we recognize in the ql, and pass them on. The raw string is stored because `SemanticQuery` is referential, and we can't clone it to owned. `branch` is a separate field so that we can pass it to tantivy queries.

It would be great if we could store an owned `SemanticQuery` or the sort so we don't need to break out the individual values (and re-construct them manually when anything changes), but I didn't have a good idea how to do that.

Reply:
In that case, can we store an owned `SemanticQuery` on the `Exchange`, in order to capture these other filters? As noted [above](https://github.com/BloopAI/bloop/pull/639#discussion_r1254099183), we know that `branch` is per-exchange. Rather than storing each filter separately on `Exchange`, this would entail storing an owned `SemanticQuery<'static>` (which encapsulates the `branch` set).

We can create an owned value of type `SemanticQuery<'static>` with a method like so:

```rust
impl SemanticQuery<'a> {
    pub fn into_owned(self) -> SemanticQuery<'static> {
        // Recursively create this method on all lifetime bearers, terminating by creating an
        // owned `Cow<'static, str>` in `Literal`.

        // ...
    }
}
```

And we can create `Cow<'static, str>` from `Cow<'a, str>` value like so:

```rust
fn example<'a>(cow: Cow<'a, str>) -> Cow<'static, str> {
    Cow::Owned(cow.into_owned())
}
```

Reply:
@calyptobai If this is a quick change can you take this on now?

Reply:
Resolved in [3d772c3](https://github.com/BloopAI/bloop/pull/639/commits/3d772c307205938ca18ea42c5ad0f34ed4ce7490)